app/src/generalpage.h

```diff
@@ -58,13 +58,16 @@ private slots:
     void backgroundChanged(QAbstractButton* button);
     void frameCacheNumberChanged(int value);
     void invertScrollDirectionBoxStateChanged(int b);
+    void newUndoRedoCheckBoxStateChanged(bool b);
 
 private:
 
     void updateSafeHelperTextEnabledState();
 
     Ui::GeneralPage* ui = nullptr;
     PreferenceManager* mManager = nullptr;
+
+    bool mInitialNewUndoSystemStateEnabled = false;
```

Comment:
Any reason to keep a member variable here?

Reply:
It's to avoid getting the "restart required" dialog when toggling the feature on and off in the same session. It's not crucial though just slightly confusing since the text is otherwise misleading.